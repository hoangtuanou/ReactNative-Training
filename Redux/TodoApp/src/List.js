import React, {Component} from 'react';
import {
	View, TouchableWithoutFeedback, Text, StyleSheet, Animated
} from 'react-native';

export default class List extends Component{


	animationAdd = () => {
		Animated.
	}

	render(){
		const {list} = this.props;
		return(
			<View>
				{
					list.map((text,i)=>{
						const {onPressItem} = this.props;
						return (
							<TouchableWithoutFeedback
								key={i}
								onPress={()=>{
										onPressItem(i);
									}
								}
							>
								<Animated.View
									ref="view"
									style={styles.item}
									animation={'lightSpeedIn'}
									delay={i*500}
								>
									<Text>{text}</Text>
								</Animated.View>
							</TouchableWithoutFeedback>
						)
					})
				}
			</View>
		)
	}
}

const styles = StyleSheet.create({
	item: {
    backgroundColor: 'whitesmoke',
    marginBottom: 5,
    padding: 15,
  }
})