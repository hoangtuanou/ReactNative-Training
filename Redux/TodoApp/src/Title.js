import React from 'react';
import {
	View,Text
} from 'react-native';

const Title = (props) => (
	<View style={{backgroundColor: 'cyan',alignItems:'center'}}>
		<Text>{props.children}</Text>
	</View>
)

export default Title;