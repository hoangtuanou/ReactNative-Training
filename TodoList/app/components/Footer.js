import React, { Component, PropTypes } from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'

export default class Footer extends Component {
  render() {
    return (
  		<TouchableOpacity style={styles.footer}>
  			<Text style={styles.text}>
  				Remove completed items
  			</Text>
  		</TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
	footer: {
		paddingVertical: 15,
		alignItems: 'center'
	},
	text: {
		color: '#CD5C5C'
	}
})
