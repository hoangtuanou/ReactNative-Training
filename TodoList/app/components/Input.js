import React, { Component, PropTypes } from 'react'
import { TextInput, View, StyleSheet } from 'react-native'

export default class Input extends Component {
  state = {text: ''}

  onChangeText = (text) => {
    this.setState({text});
  }

  onSubmitEditing = () => {
    const {onSubmit} = this.props;
    const {text} = this.state;

    if(!text) return;
    onSubmit(text);
    this.setState({text: ''});
  }

  render() {
    const {text} = this.state;
    return (
      <View style={styles.container}>
        <TextInput
          style={styles.inputText}
          underlineColorAndroid='transparent'
          placeholder='Type here, then enter'
          value={text}
          onChangeText={this.onChangeText}
          onSubmitEditing={this.onSubmitEditing}
          blurOnSubmit={false}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  inputText: {
    padding: 15,
    height: 50
  }
})
