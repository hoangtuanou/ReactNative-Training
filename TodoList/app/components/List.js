import React, { Component, PropTypes } from 'react'
import { View, Text, ScrollView, StyleSheet, TouchableOpacity } from 'react-native'

import Checkbox from './Checkbox'

export default class List extends Component {

	createItem = (item,i) => {
		const {onToggle,onRemoveItem} = this.props;
		return (
			<View
				key={i}
				style={styles.itemWrapper}
			>
				<Text style={{textDecorationLine:item.completed?'line-through':'none'}}>{item.label}</Text>
				<View style={styles.rightFeatures}>
					<Checkbox
						onToggle={()=>onToggle(i)}
						isChecked={item.completed}
					/>
					<TouchableOpacity onPress={()=>onRemoveItem(i)}>
						<Text style={styles.remove}>&times;</Text>
					</TouchableOpacity>
				</View>
			</View>
		)
	}

  render() {
		const {items} = this.props;
    return (
    	<ScrollView style={styles.listWrapper}>
    		{
					items.map(this.createItem)
				}
    	</ScrollView>
    )
  }
}

const styles = StyleSheet.create({
	listWrapper: {
		flex: 1
	},
	itemWrapper: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		borderColor: 'whitesmoke',
		borderWidth: 1,
		paddingVertical: 15,
		paddingHorizontal: 15
	},
	rightFeatures: {
		flexDirection: 'row',
		alignItems: 'center'
	},
	remove: {
		color: '#CD5C5C',
		fontSize: 26,
		marginLeft: 10
	}
})