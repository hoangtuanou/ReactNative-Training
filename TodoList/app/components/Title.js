import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'

export default class Title extends Component {

  render() {
    return (
    	<View style={styles.title}>
    		<Text style={styles.text}>
    			Todo List
    		</Text>
    	</View>
    )
  }
}

const styles = StyleSheet.create({
	title: {
		backgroundColor: 'cyan',
		padding: 15
	},
	text: {
		color: '#FFFFFF',
		textAlign: 'center'
	}
})
