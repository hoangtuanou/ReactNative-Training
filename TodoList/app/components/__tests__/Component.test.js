import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Title from '../Title';
import List from '../List';
import Footer from '../Footer';
import Input from '../Input';

describe('render correctly Title Component', () => {
  it('render correctly', () => {
    const tree = renderer.create(
      <Title/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})

describe('render correctly Input Component', () => {
  const props = {
    onSubmit: () => {}
  }
  const component = renderer.create(
    <Input {...props}/>
  );
  it('render correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('check function', () => {
    let tree = component.toJSON();
    const item = tree.children[0];

    //handle Change text
    item.props.onChangeText();

    //handle Submit
    item.props.onSubmitEditing();

    expect(tree).toMatchSnapshot();
  });
})

describe('render correctly List Component', () => {
  const props = {
    items: [{label:'sleep',completed:false},{label:'work',completed:true}],
    onToggle: () => {},
    onRemoveItem: () => {}
  };
  const component = renderer.create(
    <List {...props}/>
  );
  it('render correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('check function', () => {
    let tree = component.toJSON();

    const item = tree.children
      .filter(child => child.type === 'View')[0]
      .children.filter(child => child.type === 'View')[0].children;

    // Checkbox Function - Toggle Item
    item[0].props.onPress();

    // Remove Function
    item[1].props.onPress();

    expect(tree).toMatchSnapshot();
  })
})

describe('render correctly Footer Component', () => {
  const props ={
    onRemoveCompleted: () => {}
  };

  const component = renderer.create(
    <Footer {...props}/>
  );

  it('render correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
