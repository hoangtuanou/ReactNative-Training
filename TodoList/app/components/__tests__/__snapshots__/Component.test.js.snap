exports[`render correctly Footer Component render correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "paddingVertical": 15,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#CD5C5C",
      }
    }>
    Remove completed items
  </Text>
</TouchableOpacity>
`;

exports[`render correctly Input Component check function 1`] = `
<View
  style={undefined}>
  <TextInput
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Type here, then enter"
    style={
      Object {
        "height": 50,
        "padding": 15,
      }
    }
    underlineColorAndroid="transparent"
    value="" />
</View>
`;

exports[`render correctly Input Component render correctly 1`] = `
<View
  style={undefined}>
  <TextInput
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Type here, then enter"
    style={
      Object {
        "height": 50,
        "padding": 15,
      }
    }
    underlineColorAndroid="transparent"
    value="" />
</View>
`;

exports[`render correctly List Component check function 1`] = `
<ScrollView
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "whitesmoke",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textDecorationLine": "none",
        }
      }>
      sleep
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <View
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 2,
              "height": 20,
              "width": 20,
            }
          } />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#CD5C5C",
              "fontSize": 26,
              "marginLeft": 10,
            }
          }>
          ×
        </Text>
      </TouchableOpacity>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "whitesmoke",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textDecorationLine": "line-through",
        }
      }>
      work
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <View
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 2,
              "height": 20,
              "width": 20,
            }
          }>
          <View
            style={
              Object {
                "backgroundColor": "black",
                "flex": 1,
                "margin": 2,
              }
            } />
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#CD5C5C",
              "fontSize": 26,
              "marginLeft": 10,
            }
          }>
          ×
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</ScrollView>
`;

exports[`render correctly List Component render correctly 1`] = `
<ScrollView
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "whitesmoke",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textDecorationLine": "none",
        }
      }>
      sleep
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <View
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 2,
              "height": 20,
              "width": 20,
            }
          } />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#CD5C5C",
              "fontSize": 26,
              "marginLeft": 10,
            }
          }>
          ×
        </Text>
      </TouchableOpacity>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "whitesmoke",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textDecorationLine": "line-through",
        }
      }>
      work
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <View
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 2,
              "height": 20,
              "width": 20,
            }
          }>
          <View
            style={
              Object {
                "backgroundColor": "black",
                "flex": 1,
                "margin": 2,
              }
            } />
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#CD5C5C",
              "fontSize": 26,
              "marginLeft": 10,
            }
          }>
          ×
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</ScrollView>
`;

exports[`render correctly Title Component render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "cyan",
      "padding": 15,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFFFFF",
        "textAlign": "center",
      }
    }>
    Todo List
  </Text>
</View>
`;
